src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt

```diff
@@ -200,15 +200,22 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
         amountToHumanReadableString(
           baseTransactionWithRequestedUserReceipt.paymentNotices
             .stream()
-            .mapToInt { paymentNotice -> paymentNotice.transactionAmount().value() }
+            .mapToLong { paymentNotice -> paymentNotice.transactionAmount().value() }
             .sum())))
   }
 
+  fun amountToHumanReadableString(amount: Long): String {
+    return amountToHumanReadableString(amount.toString())
+  }
```

Comment:
we can avoid method overload changing input parameter type to Int and Long superclass [Number](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-number/)